cxx/src/lib/io/fileio.cc

```diff
@@ -129,4 +212,80 @@ size_t fread_from_file(TimeSeries& d,const string dir, const string dfile,
 		return ns_read;
 	}catch(...){throw;};
 }
+size_t fread_from_files(mspass::seismic::Ensemble<mspass::seismic::Seismogram> &de, const std::string dir, 
```

Comment:
1.  Name should be singular `fread_from_file` since only one file is referenced.
2. Change the signature to the following:
```
size_t fread_from_file(mspass::seismic::LoggingEnsemble<mspass::seismic::Seismogram>&de,
 const std::string dir, const std::string dfile)
```
See related comment in pull request page for why.  I'll try to show patches needed for this below but may miss some.

Reply:
I think there may be a case that the elements in the ensemble are stored in different files. And `fread_from_file` function reads only a part of the ensemble (several TimeSeries or Seismogram objects) from this single file. I think we need a parameter `indexes` here to specify which part to read from this file, and also specify the order, as we need the sequential reads.